### PR TITLE
#163113954  Build post a question endpoint with tests

### PR DESCRIPTION
**Pull request by Mupa1**
__What does this PR do?__
It enables a user to get a post a question to a specific meetup

__Descriptions of the tasks to be completed?__
Have the following endpoint working
```POST /api/v1/questions```

__How should this be manually tested?__
* After cloning the repo cd into it and RUN flask
* Run tests using pytest
* Using Postman test the endpoint POST /api/v1/questions to post a question

__Relevant pivotal tracker stories__
https://www.pivotaltracker.com/n/projects/2235450
#163113954 

__Screenshot__
image

### CODE REVIEW

**Comment by Ogollah** [comment]
the votes parts are not clear for me. as I think if you include the number of votes then it should be both upvotes and downvotes.

File: app/api/v1/questions/views.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            "status": 201,
+            "message" : ('Posted Question Record'),
+            "data": {
+                "id": question_id,
+                "user": request.json.get('user'),
+                "meetup": request.json.get('meetup'),
+                "title": request.json.get('title'),
+                "body":request.json.get('body'),
+            }
+        })
```
Comment (Ogollah) [inline app/api/v1/questions/views.py]: New line is recommended at the end of the file

**Comment by jkamz** [comment]
> the votes parts are not clear for me. as I think if you include the number of votes then it should be both upvotes and downvotes.

Actually I think she have implemented it correctly. Her upvote and downvote endpoints will either add or subtract from the total vote tally

**Comment by jkamz** [comment]
It looks good. Keep up.

File: app/api/v1/questions/views.py (hunk 1 above)
Comment (Mupa1) [reply app/api/v1/questions/views.py]: Thanks @Ogollah, i will correct that

**Comment by jkamz** [comment]
> > the votes parts are not clear for me. as I think if you include the number of votes then it should be both upvotes and downvotes.
> 
> Actually I think she have implemented it correctly. Her upvote and downvote endpoints will either add or subtract from the total vote tally
>I think having a different upvote and downvote entry would entail duplicating code. wouldn't it be easier to have just one entry and then either add or subtract from it depending on whether the user have upvoted or downvoted a question?

**Comment by Mupa1** [comment]
@jkamz actually, ii think @Ogollah  is right. Since this endpoint is for posting a question, i have removed the vote part completely which will be only on my upvote and downvote endpoints

**Comment by jkamz** [comment]
> @jkamz actually, ii think @Ogollah is right. Since this endpoint is for posting a question, i have removed the vote part completely which will be only on my upvote and downvote endpoints

does that mean new questions where no one has voted yet will not have a votes section? I think it was a good idea to initialize all new questions with 0 votes. The votes can then go up or down based on the voting by users as defined in your upvote and downvote methods.

**Comment by Mupa1** [comment]
@jkamz let me create a bug story to initialize the vote with zero. Thanks noting that.

**Pull request merged**